app/assets/javascripts/darkswarm/controllers/checkout/accordion_controller.js.coffee

```diff
@@ -1,6 +1,6 @@
 Darkswarm.controller "AccordionCtrl", ($scope, localStorageService, $timeout, $document, CurrentHub) ->
   key = "accordion_#{$scope.order.id}#{CurrentHub.hub.id}#{$scope.order.user_id}"
-  value = if localStorageService.get(key) then {} else { details: true, billing: false, shipping: false, payment: false }
+  value = { details: true, billing: true, shipping: true, payment: true }
```

Comment:
Why did you remove the logic of keeping the current state? If parts of the form are already filled in, we don't need to expand them.

Reply:
Sorry I don't really understand this line, looked like: if we find the key in localStorage, then the value is `{}`, otherwise we only expand details. This is the initial state of the accordion right? 

I would like to set the initial state of the accordion based on each step form being valid, so, we would only expand a step if its form is invalid (so needs the user's input). 

Could somebody guide me a bit on this? 😊 

Thanks!

Reply:
I was a bit confused with assigning `{}` when there's there's a saved state in localStorage, now I understand a bit more how it works. 😄 

I'm still working on this plus we are also discussing about some missing details, I'll push some updates in the next days!

Reply:
OK, just pushed an update. So, what I propose is: instead of storing the accordion step state in localStorage, we set the open/close initial state based on each step being valid/invalid https://github.com/openfoodfoundation/openfoodnetwork/pull/3047/files#diff-22ad100d0355fdbb60c33f3143d2a215R9

So, if you already filled up some steps and come back, the steps that doesn't need user interaction will be closed 😄

Reply:
Sounds great to me! Less complexity and more deterministic behaviour. :-D